server/stage.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/google/jsonapi"
+	"github.com/labstack/echo/v4"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+)
+
+func (s *Server) registerStageRoutes(g *echo.Group) {
+	g.PATCH("/pipeline/:pipelineID/stage/:stageID/status", func(c echo.Context) error {
+		ctx := c.Request().Context()
+		stageID, err := strconv.Atoi(c.Param("stageID"))
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Stage ID is not a number: %s", c.Param("stageID"))).SetInternal(err)
+		}
+		pipelineID, err := strconv.Atoi(c.Param("pipelineID"))
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Pipeline ID is not a number: %s", c.Param("pipelineID"))).SetInternal(err)
+		}
+
+		currentPrincipalID := c.Get(getPrincipalIDContextKey()).(int)
+		stageTasksStatusPatch := &api.StageTasksStatusPatch{
+			ID:        stageID,
+			UpdaterID: currentPrincipalID,
+		}
+		if err := jsonapi.UnmarshalPayload(c.Request().Body, stageTasksStatusPatch); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed update stage tasks status request").SetInternal(err)
+		}
+
+		tasks, err := s.store.FindTask(ctx, &api.TaskFind{PipelineID: &pipelineID, StageID: &stageID}, true /* returnOnErr */)
```

Comment:
nit:  move this to L59 (just above var tasksPatched []*api.Task)

Reply:
Done.